db/types/alteration.py

```diff
@@ -0,0 +1,209 @@
+from sqlalchemy import text, DDL, MetaData, Table
+from db.types import base, email
+
+BOOLEAN = "boolean"
+EMAIL = email.QUALIFIED_EMAIL
+INTERVAL = "interval"
+NUMERIC = "numeric"
+TEXT = "text"
+VARCHAR = "varchar"
+
+
+def get_supported_alter_column_types(engine):
+    dialect_types = engine.dialect.ischema_names
```

Comment:
Datetimes should be fairly simple to add too, right?

Reply:
I'd rather leave those for a separate issue / PR, since there'll be some complexity around dates vs times vs datetimes, and TZs (how to decide whether a column of times should be imported with the local TZ, assuming UTC, etc.)

Reply:
I know we've discussed the plan w.r.t. TZs, but I think having dates/times/datetimes separated out in a well-documented PR might benefit future code archaeologist.

Reply:
Makes sense.